src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeMaker.java

```diff
@@ -1138,26 +1138,17 @@ boolean isUnqualifiable(Symbol sym) {
             sym.owner.kind == MTH || sym.owner.kind == VAR) {
             return true;
         } else if (sym.kind == TYP && toplevel != null) {
-            Iterator<Symbol> it = toplevel.namedImportScope.getSymbolsByName(sym.name).iterator();
-            if (it.hasNext()) {
-                Symbol s = it.next();
-                return
-                  s == sym &&
-                  !it.hasNext();
-            }
-            it = toplevel.packge.members().getSymbolsByName(sym.name).iterator();
-            if (it.hasNext()) {
-                Symbol s = it.next();
-                return
-                  s == sym &&
-                  !it.hasNext();
-            }
-            it = toplevel.starImportScope.getSymbolsByName(sym.name).iterator();
-            if (it.hasNext()) {
-                Symbol s = it.next();
-                return
-                  s == sym &&
-                  !it.hasNext();
+            for (Scope scope : new Scope[] {toplevel.namedImportScope,
```

Comment:
I wonder if we could deal with this with a compound scope? (and avoid the loop)

Reply:
In principle, we could, but the `CompoundScope` is a bit heavier, so it might be better to keep the loop here.